doc/user-guide/data_model.rst

```diff
@@ -0,0 +1,251 @@
+.. _pyvista_data_model:
+
+
+PyVista Data Model
+==================
+PyVista uses the same data types as VTK, but structures them in a more
+pythonic manner for ease of use. If you'd like a background for how
+VTK structures its data, see `Introduction to VTK in Python by Kitware
+<https://vimeo.com/32232190>`_, as well as the numerous code examples
+on `Kitware's GitHub site
+<https://kitware.github.io/vtk-examples/site/>`_. An excellent
+introduction to mathematical concept implemented in VTK is provided by
+the `Discrete Differential Geometry YouTube Series
+<https://www.youtube.com/playlist?list=PL9_jI1bdZmz0hIrNCMQW1YmZysAiIYSSS>`_
+by Prof. Keenan Crane at Carnegie Melon.
+
+
+PyVista DataSet
+===============
+
+At the most fundamental level, all PyVista geometry objects are
+:ref:`ref_dataset`.  A dataset is a surface or volume in 3D space
```

Comment:
"surface or volume" sounds like something a lot more specific than what you mean. It implies connectivity. We literally just have a subset of points embedded in 3D space at this point, right?

Reply:
@adeak Agreed. Specifically at the most fundamental level, we actually have the `vtkDataObject`, which is nothing more than `vtkFieldData` and `vtkFieldArray`s, but they are the simplest thing that can be passed to filters in `VTK`. A `vtkDataSet` expands upon `vtkDataObject`s, has field data, but also has **geometry** (notion of where the data lives in 3-space) and **topology** (notion of how the data is connnected). In mathematical terms, the different subclasses of `vtkDataSet`s are "n-simplicial complexes", typically 2-simplicial complexes for surfaces and 3-simplicial complexes for volumetric data. If the data does not need to be cleaned up and is a proper immersion in 3-space, then we say it is also "manifold".

Reply:
Modified to read:

```
At the most fundamental level, all PyVista geometry classes inherit
from the :ref:`ref_dataset` class. A dataset has geometry, topology,
and attributes describing that geometry in the form of point, cell, or
field arrays.
```